enumerables.rb

```diff
@@ -0,0 +1,184 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/BlockNesting
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < length
+        if instance_of?(Array)
+          yield self[i]
+        else
+          yield(keys[i], values[i])
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < length
+        if instance_of?(Array)
+          yield(self[i], i)
+        else
+          yield(values[i], i)
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
+
+  def my_select
+    if block_given?
+      i = 0
+      new_arr = []
+      new_hash = {}
+      while i < length
+        if instance_of?(Array)
+          new_arr << self[i] if yield self[i]
+        elsif yield(keys[i], values[i])
+          new_hash[keys[i]] = values[i]
+        end
+        i += 1
+      end
+
+      return new_arr if instance_of?(Array)
+      return new_hash if instance_of?(Hash)
+
+    else
+      puts 'block was not given'
+    end
+  end
+
```

Comment:
- [ ]  enumerables #my_select returns an enumerator if no block is given
- [ ] enumerables #my_select returns an array containing all elements of enum for which the given block returns a true value::range
``` NameError: undefined local variable or method `length' for 5..50:Range ```